From 73c8b92eec8bd566ad02b56985894947047a6fec Mon Sep 17 00:00:00 2001
From: Adam Seaton <aseaton@google.com>
Date: Fri, 26 Aug 2016 21:13:20 +0000
Subject: [PATCH 2/2] Revert "Enforce calling identity before clearing."

This reverts commit 8be3a92eb0b4105a9ed748be5a937ce79145f565.

Change-Id: I10401d57239b868f8e3514f81a0e20486838e29c
(cherry picked from commit b440ceb00fd46c9233723066c680a538067fbf82)
---
 src/com/android/providers/downloads/DownloadProvider.java | 13 -------------
 1 file changed, 13 deletions(-)

diff --git a/src/com/android/providers/downloads/DownloadProvider.java b/src/com/android/providers/downloads/DownloadProvider.java
index a4bc9fe..d30018f 100644
--- a/src/com/android/providers/downloads/DownloadProvider.java
+++ b/src/com/android/providers/downloads/DownloadProvider.java
@@ -1260,19 +1260,6 @@ public final class DownloadProvider extends ContentProvider {
             logVerboseOpenFileInfo(uri, mode);
         }
 
-        // Perform normal query to enforce caller identity access before
-        // clearing it to reach internal-only columns
-        final Cursor probeCursor = query(uri, new String[] {
-                Downloads.Impl._DATA }, null, null, null);
-        try {
-            if ((probeCursor == null) || (probeCursor.getCount() == 0)) {
-                throw new FileNotFoundException(
-                        "No file found for " + uri + " as UID " + Binder.getCallingUid());
-            }
-        } finally {
-            IoUtils.closeQuietly(probeCursor);
-        }
-
         final Cursor cursor = queryCleared(uri, new String[] {
                 Downloads.Impl._DATA, Downloads.Impl.COLUMN_STATUS,
                 Downloads.Impl.COLUMN_DESTINATION, Downloads.Impl.COLUMN_MEDIA_SCANNED }, null,
-- 
2.10.0

